libcfnet/net.c

```diff
@@ -123,7 +123,7 @@ int SendTransaction(const ConnectionInfo *conn_info,
  *  @return 0 in case of socket closed, -1 in case of other error, or
  *          >0 the number of bytes read.
  */
-int ReceiveTransaction(const ConnectionInfo *conn_info, char *buffer, int *more)
+int ReceiveTransaction(ConnectionInfo *conn_info, char *buffer, int *more)
```

Comment:
The de-constification is a bit worrying. Do you really have to carry the error information in the `ConnectionInfo`?